upup/pkg/fi/cloudup/bootstrapchannelbuilder.go

```diff
@@ -451,6 +451,29 @@ func (b *BootstrapChannelBuilder) buildAddons() *channelsapi.Addons {
 		}
 	}
 
+	// @check the node-local-dns has not been disabled
+	NodeLocalDNS := b.cluster.Spec.KubeDNS.NodeLocalDNS
+	if kubeDNS.Provider == "CoreDNS" && NodeLocalDNS.Enabled {
```

Comment:
I get a sigsegv here. Guessing you need to check for more `nil`.
You should also have a validation check for that if `NodeLocalDNS.Enabled` is true, then `kubeDNS.Provider` must be `CoreDNS`. If not people may enable NodeLocalDNS, but still wonder why they don't get the addon.